opm/material/fluidsystems/blackoilpvt/WetGasPvt.hpp

```diff
@@ -605,9 +605,15 @@ class WetGasPvt
             Scalar pSat = oilVaporizationFac.xMin() + i*delta;
             Rv = saturatedOilVaporizationFactor(regionIdx, /*temperature=*/Scalar(1e30), pSat);
 
-            std::pair<Scalar, Scalar> val(Rv, pSat);
+            Pair val(Rv, pSat);
             pSatSamplePoints.push_back(val);
         }
+
+        //Prune duplicate Rv values (can occur, and will cause problems in further interpolation)
+        auto x_coord_comparator = [](const Pair& a, const Pair& b) { return a.first == b.first; };
+        auto last = std::unique(pSatSamplePoints.begin(), pSatSamplePoints.end(), x_coord_comparator);
+        pSatSamplePoints.erase(last, pSatSamplePoints.end());
```

Comment:
This code (and the live oil version) should be unit tested, perhaps as simply as adding extra (redundant) data to an existing test?